crates/typst-pdf/src/lib.rs

```diff
@@ -89,16 +89,43 @@ pub struct PdfOptions<'a> {
     /// `Auto`, a hash of the document's title and author is used instead (which
     /// is reasonably unique and stable).
     pub ident: Smart<&'a str>,
-    /// If not `None`, shall be the creation date of the document as a UTC
-    /// datetime. It will only be used if `set document(date: ..)` is `auto`.
-    pub timestamp: Option<Datetime>,
+    /// If not `None`, shall be the creation timestamp of the document. It will
+    /// only be used if `set document(date: ..)` is `auto`.
+    pub timestamp: Option<Timestamp>,
     /// Specifies which ranges of pages should be exported in the PDF. When
     /// `None`, all pages should be exported.
     pub page_ranges: Option<PageRanges>,
     /// A list of PDF standards that Typst will enforce conformance with.
     pub standards: PdfStandards,
 }
 
+/// A timestamp with timezone information.
+#[derive(Debug, Clone, Copy)]
+pub struct Timestamp {
+    /// The datetime of the timestamp.
+    pub(crate) datetime: Datetime,
+    /// If `Some`, the timezone offset from UTC in hours and minutes.
+    /// If `None`, the timestamp is in UTC.
+    pub(crate) timezone_offset: Option<(i8, i8)>,
```

Comment:
The types should be `(i8, u8)` like in pdf-writer since the timezone can just have one sign and two negative numbers make no sense. The sign the of the hour part should also apply to the minute part, like it's done in [pdf-writer](https://docs.rs/pdf-writer/latest/pdf_writer/struct.Date.html#method.utc_offset_minute). You'll need to add some logic in the constructor to convert a signed minute offset to this representation where the sign of the hours also semantically applies to the minutes.

xmp-writer takes `(i8, i8)`, but it's buggy if the second part is negative, so we'll change to `(i8, u8)` as well. But it's not a blocker for this PR, for now you can just as cast as i8 since `minutes <= 59`.

Reply:
OK, I will change it to `(i8, u8)` and try to test it with `Canada/Newfoundland` (with UTC −03:30 and UTC −02:30 DST).